sdk/search/search-documents/src/searchIndexingBufferedSenderImpl.ts

```diff
@@ -354,7 +381,10 @@ class SearchIndexingBufferedSenderImpl<T> implements SearchIndexingBufferedSende
       // raise success event
       this.emitter.emit("batchSucceeded", result);
     } catch (e) {
-      if (this.isRetryAbleError(e) && retryAttempt < DEFAULT_RETRY_COUNT) {
+      if (this.isRetryAbleError(e) && retryAttempt <= this.maxRetries) {
+        const timeToWait = Math.min(retryAttempt * this.retryDelayInMs, this.maxRetryDelayInMs);
+        const jitterValue = Math.floor(Math.random() * 1000) + 1;
```

Comment:
Jitter shouldn't allow us to exceed maxRetryDelayInMs, so we should add it to `retryAttempt * this.retryDelayInMs` instead of doing it after.

Reply:
I have modified the code as

```
const timeToWait = Math.min(
          Math.pow(2, retryAttempt * this.retryDelayInMs + jitterValue),
          this.maxRetryDelayInMs
        );
```

Reply:
Could we use the same approach as in the retry policy?

 https://github.com/Azure/azure-sdk-for-js/blob/7dfd5ccd31ff55a63b1e97f9a9c28cdb5139804e/sdk/core/core-https/src/policies/exponentialRetryPolicy.ts#L107 

It's a slight change to how jitter is calculated, but I think it's pretty understandable

Reply:
Added the same code. That was a really good pointer. Thanks